server/rpc_test.go

```diff
@@ -205,3 +222,207 @@ func TestPing(t *testing.T) {
+		if !bytes.Equal(actualPayment.Ticket.FaceValue, protoTicket.FaceValue) {
+			return false
+		}
+		if !bytes.Equal(actualPayment.Ticket.WinProb, protoTicket.WinProb) {
+			return false
+		}
+		if !bytes.Equal(actualPayment.Ticket.RecipientRandHash, protoTicket.RecipientRandHash) {
+			return false
+		}
+		if actualPayment.Ticket.SenderNonce != protoTicket.SenderNonce {
+			return false
+		}
+		if !bytes.Equal(actualPayment.Sig, protoPayment.Sig) {
+			return false
+		}
+		if !bytes.Equal(actualPayment.Seed, protoPayment.Seed) {
+			return false
+		}
+		return true
+	})
+
+	orch.On("ProcessPayment", argMatcher, manifestID).Return(nil).Once()
+
+	err = processPayment(orch, header, manifestID)
+
+	assert.Nil(t, err)
+	orch.AssertExpectations(t)
+}
+
+func TestProcessPayment_GivenErrorFromOrch_ForwardsTheError(t *testing.T) {
+	orch := &mockOrchestrator{}
+	manifestID := core.ManifestID("some manifest")
+	protoTicket := defaultTicket(t)
+	protoPayment := defaultPayment(t, protoTicket)
+	data, err := proto.Marshal(protoPayment)
+	require.Nil(t, err)
+	header := base64.StdEncoding.EncodeToString(data)
+	orch.On("ProcessPayment", mock.Anything, mock.Anything).Return(errors.New("error from mock")).Once()
+
+	err = processPayment(orch, header, manifestID)
+
+	assert.Equal(t, "error from mock", err.Error())
+}
+
+func TestGetOrchestrator_GivenValidSig_ReturnsTranscoderURI(t *testing.T) {
+	orch := &mockOrchestrator{}
+	drivers.NodeStorage = drivers.NewMemoryDriver("")
+	uri := "http://someuri.com"
+	orch.On("VerifySig", mock.Anything, mock.Anything, mock.Anything).Return(true)
+	orch.On("ServiceURI").Return(url.Parse(uri))
+	orch.On("TicketParams", mock.Anything).Return(nil)
+
+	oInfo, err := getOrchestrator(orch, &net.OrchestratorRequest{})
+
+	assert := assert.New(t)
+	assert.Nil(err)
+	assert.Equal(uri, oInfo.Transcoder)
+}
+
+func TestGetOrchestrator_GivenInvalidSig_ReturnsError(t *testing.T) {
+	orch := &mockOrchestrator{}
+	drivers.NodeStorage = drivers.NewMemoryDriver("")
+	orch.On("VerifySig", mock.Anything, mock.Anything, mock.Anything).Return(false)
+
+	oInfo, err := getOrchestrator(orch, &net.OrchestratorRequest{})
+
+	assert := assert.New(t)
+	assert.Contains(err.Error(), "sig")
+	assert.Nil(oInfo)
+}
+
+func TestGetOrchestrator_GivenValidSig_ReturnsOrchTicketParams(t *testing.T) {
+	orch := &mockOrchestrator{}
+	drivers.NodeStorage = drivers.NewMemoryDriver("")
+	uri := "http://someuri.com"
+	expectedParams := defaultTicketParams()
+	orch.On("VerifySig", mock.Anything, mock.Anything, mock.Anything).Return(true)
+	orch.On("ServiceURI").Return(url.Parse(uri))
+	orch.On("TicketParams", mock.Anything).Return(expectedParams)
+
+	oInfo, err := getOrchestrator(orch, &net.OrchestratorRequest{})
+
+	assert := assert.New(t)
+	assert.Nil(err)
+	assert.Equal(expectedParams, oInfo.TicketParams)
+}
+
+type mockOrchestrator struct {
+	mock.Mock
+}
+
+func (o *mockOrchestrator) ServiceURI() *url.URL {
+	args := o.Called()
+	return args.Get(0).(*url.URL)
```

Comment:
Might be worth being cautious with type asserting on nil here and checking for that in case this is ever used in the future to return a nil value

Reply:
haha I was being lazy here thinking if someone ran into that in the future they will notice (tests will panic) and they will fix it :) but since you mind it, I will add the quick fix :)